### ISSUE TITLE
Update vertex positions when structure is translated with the gizmo? (handling multiple structures)

### ISSUE THREAD

**Issue by rahul-mitra13** [nmwsharp/polyscope#276]
Hello! 

Polyscope is great and I've been using it quite a bit. I wanted to ask about translating structures, specifically surface meshes when I want to work with more than one. When I load up multiple structures in polyscope, it looks something like this: 

<img width="675" alt="Screenshot 2024-05-30 at 5 58 46 PM" src="https://github.com/nmwsharp/polyscope/assets/32472390/4850924f-b9cf-4f87-80dc-2ad2c304ec72">

Cause I don't want them overlayed, I translate one of the structures using the gizmo to end up with something like: 

<img width="565" alt="Screenshot 2024-05-30 at 6 00 46 PM" src="https://github.com/nmwsharp/polyscope/assets/32472390/f0efeb04-c4df-4e30-8c3e-8777dead1aa9">

However, let's say I render a curve network (stripes in this example) over the blue structure now. It shows up as the following: 

<img width="521" alt="Screenshot 2024-05-30 at 6 02 26 PM" src="https://github.com/nmwsharp/polyscope/assets/32472390/bd5eedfe-01ef-42b8-a583-398982eddda4">

which while pretty funny is not exactly what I want. So I have two questions, 

- Am I working with multiple structures appropriately i.e., is there a way I can render them off the bat so that they're not overlayed? 
- If I have to use the gizmo, is there a way I can also update the vertex positions so that the curve network doesn't get rendered in an offset manner? 

Would appreciate any help or pointers! Thanks :)

**Comment by Lucascuibu** [comment]
Hi rahul, 
I guess the mixture of blue and yellow is correctly rendered as intended. Such behaviour inheritances from opengl. I would recommend adjust the positions of two objects in other software like blender, and export each of them separately so that you maintain the relative position and still receive two independent objects.

**Comment by nmwsharp** [comment]
Hi Rahul, good questions! Indeed, the rendering artifacts in the first image are just good old Z fighting, which I assume you're familiar with.

This structure positioning behavior is determined by the `autocenterStructures` [options flag](https://polyscope.run/basics/scene_options/#autocenter-structures).  If that flag is enabled, each structure you register will be individually translated to put the center of its bounding box at the origin. If it is disabled, each structure will be rendered at exactly the world coordinates you give for it.

Neither behavior does what you want if you are registering multiple structures and want them automatically displayed next to each other :) This would be a cool feature to implement, but there's nothing to do that in Polyscope now. Your only option is to manually set appropriate offsets (either by translating the given world coordinates, or by [setting a transform programmatically as you do with the gizmo](https://polyscope.run/structures/structure_management/#transforms)).

Likewise, there is no built-in support to "parent" the curve network to the mesh and keep their transforms synced up. You have to do it manually/programmatically if you want it to happen (and you should almost certainly disable `autocenterStructures`). This would also be a great feature to add.

**Comment by nmwsharp** [comment]
I created a feature request issue here https://github.com/nmwsharp/polyscope/issues/280 for the synced transform thing, that is particularly do-able and would be very useful.

**Comment by rahul-mitra13** [comment]
Hi Nick, thanks for the detailed write-up. :) 

Yea, I looked into the autocenter flag and setting the transform manually when I encountered this problem but was wondering if there's a simpler or more efficient way polyscope could handle this. I think I'll just stick to manually fixing offsets for my purposes for the time being. I'm a little busy right now but if some time opens up, I'll try writing up the feature! 

I'll go ahead and close this issue. Thanks once again!
